wire/blockheader.go

```diff
@@ -82,6 +83,14 @@ func (h *BlockHeader) Deserialize(r io.Reader) error {
 	return readBlockHeader(r, 0, h)
 }
 
+// FromBytes deserializes a block header byte slice.
+//
+// NOTE: This utility function was ported from github.com/decred/dcrd/wire
+func (h *BlockHeader) FromBytes(b []byte) error {
```

Comment:
I realize this and `Bytes` were added to dcrd, but I really don't want to see them in btcd.  It hides the fact there are readers/writers and their associated allocations and has much different semantics than the `btcutil.{Block,Tx}.Bytes` functions which memoize in order to avoid extra work.

Further, and perhaps even more important, this would make the code inconsistent since the newly added code would be calling these functions will the rest would still be doing it the other more transparent way.  If such a function were to be added (which as I already stated above I would not prefer to see anyways), it would need to be in a separate PR that also updated all callers to make use of it so everything stays consistent.

Please kill them with prejudice.